### PR TITLE
feat(calendar): add styles for disabled and special dates

### LINKED ISSUE #98: Calendar is missing disabled and special dates styles
## Description  
Calendar is missing disabled and special dates styles.

## Steps to reproduce  

1. Open the calendar.stories.ts file.
2. Change the disabledDates and specialDates like this:
```ts
  const today = new Date();
  const disabledDates: DateRangeDescriptor[] = [
    {
      type: DateRangeType.Between,
      dateRange: [
        new Date(today.getFullYear(), today.getMonth(), 1),
        new Date(today.getFullYear(), today.getMonth(), 5),
      ],
    },
  ];
  const specialDates: DateRangeDescriptor[] = [
    {
      type: DateRangeType.Between,
      dateRange: [
        new Date(today.getFullYear(), today.getMonth(), 6),
        new Date(today.getFullYear(), today.getMonth(), 10),
      ],
    },
  ];
```

## Result  
There is not visual styling for the disabled and special dates.

## Expected result  
There should be visual styling for the disabled and special dates. Here are Angular Calendar links for reference:
https://www.infragistics.com/products/ignite-ui-angular/angular/components/calendar#disabled-dates
https://www.infragistics.com/products/ignite-ui-angular/angular/components/calendar#special-dates

### PR DESCRIPTION
**Pull request by desig9stein**
closes #98

### CODE REVIEW

File: src/components/calendar/calendar.material.scss [hunk 1]
```diff
@@ -5,12 +5,13 @@
 $date-size-large: rem(32px);
 $date-size-medium: rem(28px);
 $date-size-small: rem(24px);
-
+$calendar-bacground: color(surface);
```
Comment (simeonoff) [inline src/components/calendar/calendar.material.scss:8]:
```suggestion
$calendar-background: color(surface);
```

File: src/components/calendar/calendar.material.scss [hunk 2]
```diff
@@ -5,12 +5,13 @@
...
 
 :host {
     display: flex;
     flex-direction: column;
     font-family: var(--igc-font-family);
+    background: $calendar-bacground;
```
Comment (simeonoff) [inline src/components/calendar/calendar.material.scss:14]:
```suggestion
    background: $calendar-background;
```

**Review by simeonoff — CHANGES_REQUESTED** [review]

**Review by simeonoff — DISMISSED** [review]

**Review by StefanIvanov — APPROVED** [review]

**Comment by desig9stein** [comment]
@DiyanDimitrov @StefanIvanov  after fixing the styling I saw that I am able to navigate over the disabled dates using the keyboard. Is this on purpose? I am asking because I saw that in igniteui for angular this is not possible.

**Review by StefanIvanov — APPROVED** [review]

**Comment by StefanIvanov** [comment]
> @DiyanDimitrov @StefanIvanov after fixing the styling I saw that I am able to navigate over the disabled dates using the keyboard. Is this on purpose? I am asking because I saw that in igniteui for angular this is not possible.

This looks like a bug to me, disabled dates should not be possible to navigate to by any means because they are disabled :)

**Comment by DiyanDimitrov** [comment]
> > @DiyanDimitrov @StefanIvanov after fixing the styling I saw that I am able to navigate over the disabled dates using the keyboard. Is this on purpose? I am asking because I saw that in igniteui for angular this is not possible.
> 
> This looks like a bug to me, disabled dates should not be possible to navigate to by any means because they are disabled :)

Actually this is intended behavior in the IgcCalendar. The disabled dates can be navigated using the keyboard, but cannot be selected. When using a screen reader while navigating the disabled dates you will be notified that this date is disabled. If the disabled dates are not focusable we will just skip them without notifying the screen reader why.

**Comment by DiyanDimitrov** [comment]
> > > > @DiyanDimitrov @StefanIvanov after fixing the styling I saw that I am able to navigate over the disabled dates using the keyboard. Is this on purpose? I am asking because I saw that in igniteui for angular this is not possible.
> > > 
> > > 
> > > This looks like a bug to me, disabled dates should not be possible to navigate to by any means because they are disabled :)
> > 
> > 
> > Actually this is intended behavior in the IgcCalendar. The disabled dates can be navigated using the keyboard, but cannot be selected. When using a screen reader while navigating the disabled dates you will be notified that this date is disabled. If the disabled dates are not focusable we will just skip them without notifying the screen reader why.
> 
> ОК I didn't see that use case and it is a valid one, but then how about Angular, should we suggest this as an enhancement there?

Yes, we could implement it with all other design changes that we have made in the IgcCalendar.

**Review by simeonoff — DISMISSED** [review]

**Comment by DiyanDimitrov** [comment]
Clicking on outside days is broken, we should switch the month to previous/next month when clicking leading/trailing days.

**Comment by DiyanDimitrov** [comment]
> Clicking on outside days is broken, we should switch the month to previous/next month when clicking leading/trailing days.

It was an issue with the parts that we apply. I have fixed it.

**Review by DiyanDimitrov — APPROVED** [review]

**Pull request merged**